packages/dev/core/src/Meshes/Node/Blocks/linearClonerBlock.ts

```diff
@@ -0,0 +1,213 @@
+
+        //rotation is magnitude per step, or total rotation
+        this.registerInput("rotation", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(0, 0, 0));
+
+        //scale is magnitude per step, or total rotation
+        this.registerInput("scale", NodeGeometryBlockConnectionPointTypes.Vector3, true, new Vector3(0, 0, 0));
+
+        this.registerOutput("output", NodeGeometryBlockConnectionPointTypes.Geometry);
+    }
+
+    /**
+     * Gets the current class name
+     * @returns the class name
+     */
+    public getClassName() {
+        return "LinearClonerBlock";
+    }
+
+    /**
+     * Gets the geometry input component
+     */
+    public get geometry(): NodeGeometryConnectionPoint {
+        return this._inputs[0];
+    }
+
+    /**
+     * Gets the count input component
+     */
+    public get count(): NodeGeometryConnectionPoint {
+        return this._inputs[1];
+    }
+
+    /**
+     * Gets the origin input component
+     */
+    public get origin(): NodeGeometryConnectionPoint {
+        return this._inputs[2];
+    }
+
+    /**
+     * Gets the direction input component
+     */
+    public get direction(): NodeGeometryConnectionPoint {
+        return this._inputs[3];
+    }
+
+    /**
+     * Gets the rotation input component
+     */
+    public get rotation(): NodeGeometryConnectionPoint {
+        return this._inputs[4];
+    }
+
+    /**
+     * Gets the scale input component
+     */
+    public get scale(): NodeGeometryConnectionPoint {
+        return this._inputs[5];
+    }
+
+    /**
+     * Gets the geometry output component
+     */
+    public get output(): NodeGeometryConnectionPoint {
+        return this._outputs[0];
+    }
+
+    protected _buildBlock(state: NodeGeometryBuildState) {
+        const func = (state: NodeGeometryBuildState) => {
+            state.pushExecutionContext(this);
+
+            this._vertexData = this.geometry.getConnectedValue(state);
+            state.pushGeometryContext(this._vertexData);
+
+            if (!this._vertexData || !this._vertexData.positions || !this.geometry.isConnected) {
+                state.restoreExecutionContext();
+                state.restoreGeometryContext();
+                this.output._storedValue = null;
+                return;
+            }
+            const count = this.count.getConnectedValue(state) as number;
+
+            if (count <= 0) {
+                this.output._storedValue = this._vertexData;
+                state.restoreExecutionContext();
+                state.restoreGeometryContext();
+            }
+
+            const origin = this.origin.getConnectedValue(state) as Vector3;
+            const direction = this.direction.getConnectedValue(state) as Vector3;
+            const rotation = this.rotation.getConnectedValue(state) as Vector3;
+            const scale = this.scale.getConnectedValue(state) as Vector3;

```

Comment:
as scale can be a float:
https://github.com/BabylonJS/Babylon.js/blob/58a04329c4d1f591d4c783c694541a7dff0e62e6/packages/dev/core/src/Meshes/Node/Blocks/Instances/instantiateBlock.ts#L158